src/modules/aws_ec2/entity/instance_block_device_mapping.ts

```diff
@@ -0,0 +1,102 @@
+import { Column, Entity, Index, JoinColumn, ManyToOne, PrimaryGeneratedColumn, Unique } from 'typeorm';
+
+import { cloudId } from '../../../services/cloud-id';
+import { AwsRegions } from '../../aws_account/entity';
+import { GeneralPurposeVolume } from './general_purpose_volume';
+import { Instance } from './instance';
+
+/**
+ * Table to manage AWS EC2 block device mappings. Each instance that you launch has an associated root device volume,
+ * which is either an Amazon EBS volume or an instance store volume. You can use block device mapping to specify
+ * additional EBS volumes or instance store volumes to attach to an instance when it's launched.
+ *
+ * @see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/block-device-mapping-concepts.html
+ *
+ */
+@Entity()
+export class InstanceBlockDeviceMapping {
+  /**
+   * @private
+   * Auto-incremented ID field
+   */
+  @PrimaryGeneratedColumn()
+  id?: number;
+
+  /**
+   * @public
+   * Device name to associate this volume to the instance
+   */
+  @Column()
+  deviceName: string;
+
+  @Column({ nullable: false })
+  instanceId?: number;
+
+  /**
+   * @public
+   * Internal AWS ID for the instance
+   */
+  @Column({
+    nullable: true,
+    comment: 'Unique identifier provided by AWS once the instance is provisioned',
+  })
+  @cloudId
+  cloudInstanceId?: string;
+
+  @Column({ nullable: true })
+  @Index({ unique: true, where: 'volume_id IS NOT NULL' })
+  volumeId?: number;
+
+  /**
+   * @public
+   * Internal AWS ID for the volume
+   */
+  @Column({
+    nullable: true,
+    comment: 'Unique identifier provided by AWS once the volume is provisioned',
+  })
+  @cloudId
+  cloudVolumeId?: string;
```

Comment:
i found that i actually needed to have some @cloudId on the instance_block_device_mapping entity itself. Otherwise it was not able to reconciliate between cloud and database. And what is representing a set of valid cloudId here is cloudInstanceId+cloudVolumeId+region.
It is different than the instanceId and volumeId defined here, as those values are pointing to database ids, while the cloud ones are pointing to aws identifiers